### PR TITLE
remove image-registry buckets on destroy

### PR DESCRIPTION
**Pull request by sjenning**
@ironcladlou

The image registry operator in the guest cluster creates an s3 bucket for storage after https://github.com/openshift/hypershift/pull/187

However, these buckets are not currently removed during `destroy {cluster|infra}`.

This PR adds remove buckets that have the prefix `<infraid>-image-registry-` during infra destroy.

### CODE REVIEW

File: test/e2e/e2e_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
+// +build e2e
```
Comment (ironcladlou) [inline test/e2e/e2e_suite_test.go:1]: Not sure you intended to add this file, you might need to rebase

File: test/e2e/quick_start_test.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
+// +build e2e
```
Comment (ironcladlou) [inline test/e2e/quick_start_test.go:1]: Not sure you intended to add this file, you might need to rebase

**Review by ironcladlou — CHANGES_REQUESTED** [review]
Looks like a couple unwanted local files crept in to the commit

**Comment by sjenning** [comment]
@ironcladlou indeed :man_facepalming: I reverted https://github.com/openshift/hypershift/pull/250 to get a start and those files have since been removed.

**Comment by ironcladlou** [comment]
/lgtm
/hold

Just waiting on e2e

**Pull request merged**